index.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+/* eslint-disable no-underscore-dangle */
+const Joi = require('joi');
+
+class ScmBase {
+    /**
+     * Constructor for Scm
+     * @method constructor
+     * @param  {Object}    config Configuration
+     * @return {ScmBase}
+     */
+    constructor(config) {
+        this.config = config;
+    }
+
+    /**
+     * Reload configuration
+     * @method configure
+     * @param  {Object}     config      Configuration
+     */
+    configure(config) {
+        this.config = config;
+    }
+
+    /**
+     * Get a users permissions on a repository
+     * @method getPermissions
+     * @param  {Object}   config            Configuration
+     * @param  {String}   config.scmUrl     The scmUrl to get permissions on
+     * @param  {String}   config.user       The user to get pemissions for
+     * @param  {String}   config.token      The token used to authenticate to the SCM
+     * @return {Promise}
+     */
+    getPermissions(config) {
+        const result = Joi.validate(config, {
+            scmUrl: Joi.string().required(),
```

Comment:
You can probably use this for scmUrl since it's already there
https://github.com/screwdriver-cd/data-schema/blob/b98578d55f40a7c6de7c195686bdc509dde47473/config/regex.js#L17